tagcompare/capture.py

```diff
@@ -14,11 +17,164 @@
 LOGGER = logger.Logger(name="capture", writefile=True).get()
 
 
-def __capture_tags_for_configs(cids, pathbuilder,
-                               configs,
-                               tagsizes=settings.DEFAULT.tagsizes,
-                               tagtypes=settings.DEFAULT.tagtypes,
-                               capture_existing=False):
+class TagCapture(object):
+    """TagCapture uses webdriver to capture tags for campaigns"""
+
+    def __init__(self, configname, driver, caps=None):
+        self._configname = configname
+        self._driver = driver
+        self._caps = caps
+
+    def stop(self):
+        if self._driver:
+            self._driver.quit()
+
+    @classmethod
+    def from_config(cls, configname, buildname=None):
+        if configname == 'phantomjs':
+            driver = webdriver.setup_webdriver(
+                drivertype=WebDriverType.PHANTOM_JS)
+            caps = configname
+        else:
+            caps = TagCapture.__get_capabilities_for_config(
+                configname, buildname)
+            driver = webdriver.setup_webdriver(drivertype=WebDriverType.REMOTE,
+                                               capabilities=caps)
+        return cls(configname, driver, caps)
+
+    @classmethod
+    def from_caps(cls, caps):
+        browsername = caps['browserName']
+        browserversion = caps.get('version') or ''
+        configname = browsername + browserversion
+        driver = webdriver.setup_webdriver(drivertype=WebDriverType.REMOTE,
+                                           capabilities=caps)
+        return cls(configname, driver, caps)
+
+    def capture_tag(self, tag_html, output_path, tagtype='iframe'):
+        """
+        Generic/public method to capture a tag
+        :param driver:
+        :param tag_html:
+        :param output_path:
+        :param tagtype:
+        :return:
+        """
+        try:
+            errors = webdriver.display_tag(self._driver, tag_html)
+            tag_element = self._driver.find_element_by_tag_name(tagtype)
+            webdriver.screenshot_element(
+                self._driver, tag_element, output_path)
+        except selenium.common.exceptions.WebDriverException:
+            LOGGER.exception("Exception while capturing tags!")
+            return False
```

Comment:
Returning `False` on errors is not very pythonic. Why not just raise an exception? Either let this one be raised and log it on the way out, just let in be raised and don't even try to handle it here, or catch this exception and raise another one.